x-pack/test/detection_engine_api_integration/security_and_spaces/group1/get_rules_info.ts

```diff
@@ -0,0 +1,122 @@
+  const es = getService('es');
+  const log = getService('log');
+
+  describe('get_rules_info', () => {
+    beforeEach(async () => {
+      await createSignalsIndex(supertest, log);
+    });
+
+    afterEach(async () => {
+      await deleteSignalsIndex(supertest, log);
+      await deleteAllAlerts(supertest, log);
+      await deleteAllTimelines(es);
+    });
+
+    it('should return expected JSON keys', async () => {
+      const { body } = await supertest
+        .get(RULES_INFO_URL)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(200);
+
+      expect(Object.keys(body)).to.eql([
+        'rules_custom_count',
+        'rules_prebuilt_installed_count',
+        'tags',
+      ]);
+    });
+
+    it('should return the correct result when there are no rules', async () => {
+      const { body } = await supertest
+        .get(RULES_INFO_URL)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(200);
+      expect(body.rules_custom_count).to.eql(0);
+      expect(body.rules_prebuilt_installed_count).to.eql(0);
+      expect(body.tags).to.eql([]);
+    });
+
+    describe('when there is a custom rule', () => {
+      beforeEach(async () => {
+        const rule = getSimpleRule();
+        rule.tags = ['tag-a'];
+
+        await supertest
+          .post(DETECTION_ENGINE_RULES_URL)
+          .set('kbn-xsrf', 'true')
+          .send(rule)
+          .expect(200);
+      });
+
+      it('should return the correct number of custom rules', async () => {
+        const { body } = await supertest
+          .get(RULES_INFO_URL)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+        expect(body.rules_custom_count).to.eql(1);
+        expect(body.rules_prebuilt_installed_count).to.eql(0);
+      });
+
+      it('should return correct tags', async () => {
+        const { body } = await supertest
+          .get(RULES_INFO_URL)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+        expect(body.tags).to.eql(['tag-a']);
+      });
+    });
+
+    describe('when there are installed prebuilt rules', () => {
+      beforeEach(async () => {
+        await supertest.put(PREBUILT_RULES_URL).set('kbn-xsrf', 'true').send().expect(200);
+      });
+
+      it('should return the correct number of installed prepacked rules after pre-packaged rules have been installed', async () => {
+        const { body } = await supertest
+          .get(RULES_INFO_URL)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+        expect(body.rules_prebuilt_installed_count).to.be.greaterThan(0);
+        expect(body.rules_custom_count).to.eql(0);
+      });
+
+      it('should return correct tags', async () => {
+        const { body } = await supertest
+          .get(RULES_INFO_URL)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+        expect(body.tags.length).to.be.greaterThan(0);
```

Comment:
Not sure I understand what that assertion tests. The title says, "should return correct tags", but the body only checks for the tags' length.

Reply:
The test has been updated